modules/core/src/main/java/org/apache/ignite/cache/query/VectorQuery.java

```diff
@@ -0,0 +1,234 @@
+ * </pre>
+ *
+ * @see IgniteCache#query(Query)
+ */
+public final class VectorQuery<K, V> extends Query<Cache.Entry<K, V>> {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** */
+    private String type;
+
+    /** */
+    private String field;
+
+    /** SQL clause. */
+    private String cause;
+
+    /** SQL clause as vector. */
+    private float[] clauseVector;
+
+    /** [K]NN, how many vectors to return. */
+    private int k;
+
+    /**
+     * Constructs query for the given search string.
+     *
+     * @param type Type.
+     * @param field Type.
+     * @param cause Search string.
+     * @param k The nuber of vectors to return.
+     */
+    public VectorQuery(Class<?> type, String field, String cause, int k) {
+        setType(type);
+        setField(field);
+        setCause(cause);
+        setK(k);
+    }
+
+    /**
+     * Constructs query for the given search string.
+     *
+     * @param type Type.
+     * @param field Type.
+     * @param clauseVector Search string.
+     * @param k The nuber of vectors to return.
+     */
+    public VectorQuery(Class<?> type, String field, float[] clauseVector, int k) {
+        setType(type);
+        setField(field);
+        setCauseVector(clauseVector);
+        setK(k);
+    }
+
+    /**
+     * Gets type for query.
+     *
+     * @return Type.
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Sets type for query.
+     *
+     * @param type Type.
+     * @return {@code this} For chaining.
+     */
+    public VectorQuery<K, V> setType(Class<?> type) {
+        this.type = QueryUtils.typeName(type);
+
+        return this;
+    }
+
+    /**
+     * Sets k.
+     *
+     * @param k K.
+     * @return {@code this} For chaining.
+     */
+    public VectorQuery<K, V> setK(int k) {
+        this.k = k;
+
+        return this;
+    }
+
+    /**
+     * Gets k.
+     *
+     * @return K.
+     */
+    public int getK() {
+        return k;
+    }
+
+    /**
+     * Gets field for query.
+     *
+     * @return Type.
+     */
+    public String getField() {
+        return field;
+    }
+
+    /**
+     * Sets field for query.
+     *
+     * @param field field.
+     * @return {@code this} For chaining.
+     */
+    public VectorQuery<K, V> setField(String field) {
+        this.field = field;
+
+        return this;
+    }
+
+    /**
+     * Gets text search string.
+     *
+     * @return Text search string.
+     */
+    public String getCause() {
+        return cause;
+    }
+
+    /**
+     * Sets text search string.
+     *
+     * @param cause Text search string.
+     * @return {@code this} For chaining.
+     */
+    public VectorQuery<K, V> setCause(String cause) {
+        A.notNull(cause, "cause");
+
+        this.cause = cause;
+
+        return this;
+    }
+
+    /**
+     * Gets text search string.
+     *
+     * @return Text search string as vector.
+     */
+    public float[] getCauseVector() {
```

Comment:
`getClauseVector()`?

Reply:
The method is still called `getCauseVector()`. Please check all occurrences of the string `cause` in this PR.